art/utils.py

```diff
@@ -630,12 +636,18 @@ def compute_success_array(
     :param batch_size: Batch size.
     :return: Percentage of successful adversarial samples.
     """
-    adv_preds = np.argmax(classifier.predict(x_adv, batch_size=batch_size), axis=1)
+    preds = classifier.predict(x_clean, batch_size=batch_size)
+    adv_preds = classifier.predict(x_adv, batch_size=batch_size)
+    if len(preds.shape) >= 2:
+        adv_preds = np.argmax(adv_preds, axis=1)
+        preds = np.argmax(preds, axis=1)
+    else:
+        adv_preds = np.round(adv_preds)
+        preds = np.round(preds)
     if targeted:
         attack_success = adv_preds == np.argmax(labels, axis=1)
-    else:
```

Comment:
We need to make sure that `attack_success` for `targeted` does not get overwritten later in line 650.

Reply:
Tentatively added back else to avoid overwrite of var, waiting for pytest results...

```
    if targeted:
        attack_success = adv_preds == np.argmax(labels, axis=1)
    else:
        attack_success = adv_preds != preds
```